source/administration/plugins.rst

```diff
@@ -77,12 +77,163 @@ Plugins may have one or both of the following parts:
+  What keysize do you want? (2048) 3072
+
+  Requested keysize is 3072 bits
+
+  Please specify how long the key should be valid.
+        0 = key does not expire
+        <n>  = key expires in n days
+        <n>w = key expires in n weeks
+        <n>m = key expires in n months
+        <n>y = key expires in n years
+  Key is valid for? (0) 0
+
+  Key expires at ...
+
+  Is this correct? (y/N) y
+
+  GnuPG needs to construct a user ID to identify your key.
+  Real name: Mattermost Inc
+
+  Email address: info@mattermost.com
+  Comment:
+
+  You selected this USER-ID:
+      "Mattermost Inc <info@mattermost.com>"
+  Change (N)ame, (C)omment, (E)mail or (O)kay/(Q)uit? O
+
+
+.. note::
+ Key size should be at least 3072 bits.
+
+Exporting the Private Key
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Find the ID of your private key first. The ID is a hexadecimal number.
+
+.. code-block:: sh
+
+  gpg --list-secret-keys
+
+This is your private key and should be kept secret. Your hexadecimal key ID will, of course, be different.
+
+.. code-block:: sh
+
+  gpg --export-secret-keys F3FACE45E0DE642C8BD6A8E64C7C6562C192CC1F > ./my-priv-key
+
+Exporting the Public Key
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Find the ID of your public key first. The ID is a hexadecimal number.
+
+.. code-block:: sh
+
+  gpg --list-keys
+
+.. code-block:: sh
+
+  gpg --export F3FACE45E0DE642C8BD6A8E64C7C6562C192CC1F > ./my-pub-key
+
+Importing the Key
+~~~~~~~~~~~~~~~~~
+
+If you already have a public and private key pair, you can import them to the GPG.
+
+.. code-block:: sh
+
+ gpg --import ./my-priv-gpg-key
+
+ gpg --import ./my-pub-gpg-key
+
+Running Plugin Signing
+^^^^^^^^^^^^^^^^^^^^^^
+
+For plugin signing, you have to know the hexadecimal ID of the private key. Let's assume you want to sign `com.mattermost.demo-plugin-0.1.0.tar.gz` file, run:
+
+.. code-block:: sh
+
+  gpg -u F3FACE45E0DE642C8BD6A8E64C7C6562C192CC1F --verbose --personal-digest-preferences SHA256 --detach-sign com.mattermost.demo-plugin-0.1.0.tar.gz
+
+This command will generate `com.mattermost.demo-plugin-0.1.0.tar.gz.sig`, which is the signature of your plugin.
+
+Plugin Verification
+^^^^^^^^^^^^^^^^^^^
+
+Mattermost server will verify plugin signatures downloaded from plugin marketplace. To add custom public keys, run the following command on the Mattermost server:
+
+.. code-block:: sh
+
+  mattermost plugin add key my-pub-key
+
+Multiple public keys can be added to the Mattermost server:
+
+.. code-block:: sh
+
+  mattermost plugin add key my-pk-file1 my-pk-file2
+
+
+To list the names of all public keys installed on your Mattermost server, use:
+
+.. code-block:: sh
+
+  mattermost plugin keys
+
+To delete public key(s) from your Mattermost server, use:
+
+.. code-block:: sh
+
+  mattermost plugin delete key my-pk-file1 my-pk-file2
+
+Implementation
+~~~~~~~~~~~~~~~~~
+
+See the _`implementation document <https://docs.google.com/document/d/1qABE7VEx4k_ZAeh6Ydn4pGbu6BQfZt65x68i2s65MOQ/>`_ for more information.
```

Comment:
The link is not formatted correctly yet - let me know if you need help.

Reply:
Thanks, @amyblais. Looks like I fixed it now.